File: android_mobile/src/main/java/com/alexstyl/specialdates/events/peopleevents/PeopleEventsViewRefresher.java
package com.alexstyl.specialdates.events.peopleevents;

import com.alexstyl.specialdates.PeopleEventsView;

import java.util.Set;

public final class PeopleEventsViewRefresher {

    private final Set<PeopleEventsView> views;

    PeopleEventsViewRefresher(Set<PeopleEventsView> views) {
        this.views = views;
    }

    public void updateAllViews() {
        for (PeopleEventsView view : views) {
            view.requestUpdate();
        }
    }
}

File: android_mobile/src/main/java/com/alexstyl/specialdates/events/peopleevents/PeopleEventsViewRefresher.kt
package com.alexstyl.specialdates.events.peopleevents

import com.alexstyl.specialdates.PeopleEventsView

class PeopleEventsViewRefresher(private val views: MutableSet<PeopleEventsView>) {

    fun refreshViews() {
        for (view in views) {
            view.onEventsUpdated()
        }
    }

    fun addView(view: PeopleEventsView) {
        this.views.add(view)
    }

    fun removeView(view: PeopleEventsView) {
        this.views.remove(view)
    }
}
